File: AccuPay/Domain/Paystubs/AllowancePerDay.vb
Option Strict On

Imports System.ComponentModel.DataAnnotations
Imports System.ComponentModel.DataAnnotations.Schema

Namespace Global.AccuPay.Entity

    <Table("allowanceperday")>
    Public Class AllowancePerDay

        <Key>
        <DatabaseGenerated(DatabaseGeneratedOption.Identity)>
        Public Overridable Property RowID As Integer?

        Public Overridable Property [Date] As Date

        Public Overridable Property Amount As Decimal

        <ForeignKey("AllowanceItemID")>
        Public Overridable Property AllowanceItem As AllowanceItem

        Protected Sub New()
        End Sub

        Public Sub New([date] As Date, amount As Decimal)
            Me.Date = [date]
            Me.Amount = amount
        End Sub

    End Class

End Namespace

File: AccuPay.Core/Entities/AllowancePerDay.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace AccuPay.Core.Entities
{
    [Table("allowanceperday")]
    public class AllowancePerDay
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public virtual int? RowID { get; set; }

        public virtual DateTime Date { get; set; }

        public virtual decimal Amount { get; set; }

        [ForeignKey("AllowanceItemID")]
        public virtual AllowanceItem AllowanceItem { get; set; }

        protected AllowancePerDay()
        {
        }

        public AllowancePerDay(DateTime date, decimal amount)
        {
            this.Date = date;
            this.Amount = amount;
        }
    }
}